src/service/history-impl.js

```diff
@@ -143,17 +143,18 @@ export class History {
   }
 
   /**
-   * Requests navigation one step back. This request is only satisifed
-   * when the history has at least one step to go back in the context
-   * of this document.
+   * Requests navigation one step back. This first attempts to go back within
+   * the context of this document.
+   *
+   * @param {boolean=} canPerformWindowBack
    * @return {!Promise}
    */
-  goBack() {
+  goBack(canPerformWindowBack) {
     return this.enque_(() => {
-      if (this.stackIndex_ <= 0) {
-        // Nothing left to pop.
+      if (this.stackIndex_ <= 0 && !canPerformWindowBack) {
```

Comment:
For `HistoryBindingVirtual`, do viewers face-plant if they receive a negative `stackIndex` param?

Reply:
The SRP won't face-plant. I believe it'll just drop the request as invalid